.github/workflows/main.yml

```diff
@@ -518,3 +518,8 @@ jobs:
       if: startsWith(github.ref, 'refs/tags/')
       with:
         files: "gh-release/*.tar.gz*"
+    - uses: dtolnay/rust-toolchain@stable
+    - run: rustc publish.rs
+    - run: ./publish publish
+      env:
+        CRATES_IO_TOKEN: ${{ secrets.CRATES_IO_TOKEN }}
```

Comment:
I think you need to copy the `if: startsWith(github.ref, 'refs/tags/')` from the above GitHub release step down to here. Right now I think this will trigger a publish on every push to `main`.

Reply:
Done.